oak_functions/client/rust/src/rekor.rs

```diff
@@ -0,0 +1,181 @@
+pub struct LogEntry {
+    #[serde(rename = "body")]
+    pub body: String,
+
+    #[serde(rename = "integratedTime")]
+    pub integrated_time: usize,
+
+    // TODO(#2316): should this be verified?
+    /// This is the SHA256 hash of the DER-encoded public key for the log at the time the entry was
+    /// included in the log
+    /// Pattern: ^[0-9a-fA-F]{64}$
+    #[serde(rename = "logID")]
+    pub log_id: String,
+
+    /// Minimum: 0
+    #[serde(rename = "logIndex")]
+    pub log_index: usize,
+
+    /// Includes a signature over the body, integratedTime, logID, and logIndex.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde(rename = "verification")]
+    pub verification: Option<LogEntryVerification>,
+}
+
+/// Struct representing a verification object in a Rekor LogEntry. The verification object in Rekor
+/// also contains an inclusion proof. Since we currently don't verify the inclusion proof in the
+/// client, it is omitted from this struct.
+#[derive(Debug, Deserialize, PartialEq, Serialize)]
+pub struct LogEntryVerification {
+    // Base64-encoded signature over the body, integratedTime, logID, and logIndex.
+    #[serde(rename = "signedEntryTimestamp")]
+    pub signed_entry_timestamp: String,
+}
+
+/// Convenient struct for verifying the `signedEntryTimestamp` in a Rekor LogEntry.
+///
+/// This bundle can be verified using the public key from Rekor. The public key can
+/// be obtained from the `/api/v1/log/publicKey` Rest API. For `sigstore.dev`, it is a PEM-encoded
+/// x509/PKIX public key. The PEM-encoded content can be parsed into a `p256::ecdsa::VerifyingKey`
+/// using `unmarshal_pem_to_p256_public_key`.
+pub struct RekorSignatureBundle {
+    /// Canonicalized JSON representation of a subset of a Rekor LogEntry fields that are signed
+    /// to generate `signedEntryTimestamp` (also a field in the Rekor LogEntry). These fields
+    /// include body, integratedTime, logID and logIndex.
+    pub canonicalized: Vec<u8>,
+
+    /// Signature over the canonicalized JSON document. This is obtained by decoding the
+    /// Base64-encoded `signedEntryTimestamp` in the Rekor LogEntry.
+    pub signature: p256::ecdsa::Signature,
+}
+
+/// Convertor for creating a RekorSignatureBundle from a Rekor LogEntry as described in https://github.com/sigstore/rekor/blob/4fcdcaa58fd5263560a82978d781eb64f5c5f93c/openapi.yaml#L433-L476.
+impl TryFrom<&LogEntry> for RekorSignatureBundle {
+    type Error = anyhow::Error;
+
+    fn try_from(log_entry: &LogEntry) -> anyhow::Result<Self> {
+        // Create a copy of the LogEntry, but skip the verification.
+        let entry_subset = LogEntry {
+            body: log_entry.body.clone(),
+            integrated_time: log_entry.integrated_time,
+            log_id: log_entry.log_id.clone(),
+            log_index: log_entry.log_index,
+            verification: None,
+        };
+
+        // Canonicalized JSON document that is signed.
```

Comment:
What constitutes canonicalization? Only that some fields are removed? Does the order of the fields make a difference?